apc_cache.c

```diff
@@ -1218,6 +1218,9 @@ PHP_APCU_API zend_bool apc_cache_defense(apc_cache_t *cache, zend_string *key, t
 PHP_APCU_API void apc_cache_serializer(apc_cache_t* cache, const char* name) {
 	if (cache && !cache->serializer) {
 		cache->serializer = apc_find_serializer(name);
+		if (strcmp(name, "default") != 0 && !cache->serializer) {
+			php_error_docref(NULL, E_WARNING, "apc_cache_serializer: serializer \"%s\" is not supported", name);
```

Comment:
Not a maintainer but I'm somewhat familiar with the codebase

`not supported or not enabled`, maybe. There's also the question of whether it should fall back to the "php" serializer instead, given that the actual default ini value is "php" (see linked ticket)

Reply:
Indeed. The current behavior is complicated and cannot be understood without following the implementation. Also, the situation where the fallback behavior that occurs when it is not present and the flow when php is explicitly specified are separated is probably not desirable.

I have added a comment to your Issue.